### ISSUE TITLE
Adding new NavigationPolygon freezes Godot.

### ISSUE THREAD

**Issue by davismiculis** [godotengine/godot#12280]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
macOS 10.12.6 (Sierra)
Godot v3.0.alpha.custom_build.2e212a2

**Issue description:**
Adding a new `NavigationPolygon` to `NavigationPolygonInstance` freezes (only force quit works) Godot if under `Sprite` node. Well, obviously, it should create blank `NavigationPolygon` sub-resource and work fine. Everything works fine if not under`Sprite` node. (Perhaps related to #11975)

<img width="343" alt="aa" src="https://user-images.githubusercontent.com/5004001/31852476-ebf3517c-b680-11e7-920c-c493ac35bbc3.png">

**Steps to reproduce:**
Create tree:
```
- Node
  - Sprite
    - NavigationPolygonInstance
```
Select `NavigationPolygonInstance` and click on `Navpoly` property and create `new NavigationPolygon`

**Link to minimal example project:**
[TEST.zip](https://github.com/godotengine/godot/files/1404156/TEST.zip)

**Comment by akien-mga** [comment]
CC @bojidar-bg

**Comment by bojidar-bg** [comment]
Unfortunately, I can't fix it right now, though a backtrace would be welcome.

**Comment by davismiculis** [comment]
This is all I could get by force closing app. I have zero experience debugging C++ software.
```
Process 19101 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGTERM
    frame #0: 0x00007fffaf030418 libc++abi.dylib`__cxxabiv1::__si_class_type_info::search_above_dst(__cxxabiv1::__dynamic_cast_info*, void const*, void const*, int, bool) const + 10
libc++abi.dylib`__cxxabiv1::__si_class_type_info::search_above_dst:
->  0x7fffaf030418 <+10>: pushq  %r12
    0x7fffaf03041a <+12>: pushq  %rbx
    0x7fffaf03041b <+13>: pushq  %rax
    0x7fffaf03041c <+14>: movl   %r9d, %r13d
```

**Comment by akien-mga** [comment]
Fixed by #12413.
